dataactcore/models/jobTrackerInterface.py

```diff
@@ -229,6 +230,62 @@ def checkJobDependencies(self,jobId):
                 # add to the job queue
                 jobQueueResult = self.jobQueue.enqueue.delay(depJobId)
 
+    def runChecks(self,jobId):
+        """ Run all checks on this jobId
+
+        Args:
+        jobId -- ID of job to be checked
+        Returns:
+        True if all passed, otherwise False or exception
+        """
+        if(self.checkJobReady(jobId) and self.checkPrerequisites(jobId)):
+            # All passed
+            return True
+        else:
+            return False
+
+    def checkJobReady(self, jobId):
+        """ Check that the jobId is located in job table and that status is ready
+        Args:
+        jobId -- ID of job to be run
+
+        Returns:
+        True if job is ready, False otherwise
+        """
+        query = self.session.query(Job.job_status_id).filter(Job.job_id == jobId)
+        result = self.checkJobUnique(query)
+        # Found a unique job
+        if(result.job_status_id != self.getJobStatusId("ready")):
+            # Job is not ready
+            # Job manager is not yet implemented, so for now doesn't have to be ready
+            return True
+            # TODO when job manager exists, change to exception below
+            #exc = ResponseException("Job is not ready",StatusCode.CLIENT_ERROR,None,ValidationError.jobError)
+            #exc.status = 400
+            #raise exc
+
+        return True
+
+    def checkPrerequisites(self, jobId):
+        """ Checks that specified job has no unsatisfied prerequisites
+        Args:
+        jobId -- job_id of job to be run
+
+        Returns:
+        True if prerequisites are satisfied, False otherwise
+        """
+        # Get list of prerequisites
+        queryResult = self.session.query(JobDependency).filter(JobDependency.job_id == jobId).all()
+        for prereq in queryResult:
+            query = self.session.query(Job).filter(Job.job_id == prereq.prerequisite_id)
```

Comment:
It looks like our ORM models aren't using Sqlalchemy's [relationships](http://docs.sqlalchemy.org/en/latest/orm/tutorial.html#building-a-relationship), which would make this a bit simpler. Rather than crafting our own queries, relationships push that logic into the ORM.

Not worth holding anything back, but worth considering.